packages/express-did-auth/test/challenge-verifier.test.ts

```diff
@@ -0,0 +1,149 @@
+import { ErrorCodes } from '../src/errors'
+import ChallengeVerifier from '../src/classes/challenge-verifier'
+import MockDate from 'mockdate'
+import { keccak256 } from 'js-sha3'
+import {
+  modulo0Timestamp, modulo8Timestamp, otherSlotTimestamp, modulo59Timestamp
+} from './utils'
+
+describe('ChallengeVerifier', () => {
+  const challengeSecret = 'theSecret'
+  const challengeExpirationTimeInSeconds = 60
+
+  const did = 'did:ethr:rsk:testnet:0xce83da2a364f37e44ec1a17f7f453a5e24395c70'
+
+  const calculateExpectedChallenge = (did: string, expirationTimeInSeconds: number, secret: string, currentTimestamp?: number) => {
+    const now = currentTimestamp || Date.now()
+    const timestamp = Math.floor(now / (expirationTimeInSeconds * 1000))
+    return keccak256(`${did}-${secret}-${timestamp}`)
+  }
+
+  afterEach(() => MockDate.reset())
+
+  describe('get', () => {
+    it('should throw an error if no did', () => {
+      const verifier = new ChallengeVerifier({ challengeSecret })
+
+      expect(() => verifier.get(undefined)).toThrow(ErrorCodes.INVALID_DID)
+    })
+
+    it('should get a challenge', () => {
+      const verifier = new ChallengeVerifier({ challengeSecret, challengeExpirationTimeInSeconds })
+
+      const actual = verifier.get(did)
+      expect(actual).toBeTruthy()
+
+      const expected = calculateExpectedChallenge(did, challengeExpirationTimeInSeconds, challengeSecret)
+      expect(actual).toEqual(expected)
+    })
+
+    it('should get the same challenge when invoking it within the same timeslot', () => {
```

Comment:
Do we really want to test this? It is white box. We just need to return a challenge that is randomised over time. The fact that our implementation returns the same random nombre during the _expiration time_ is not part of the specs

Reply:
Is not part of the specs but it is part of our implementation, this is testing the implementation of the challenge verifier, not the specs. If we implement something, we definitely test it to behave as expected